lib/auth/google_oauth2_authenticator.rb

```diff
@@ -21,6 +21,19 @@ def after_authenticate(auth_hash)
     if !result.user && !result.email.blank? && result.email_valid
       result.user = User.find_by_email(result.email)
       if result.user
+        # we've matched an existing user to this login attempt...
+        if result.user.google_user_info && result.user.google_user_info.google_user_id != google_hash[:google_user_id]
```

Comment:
With Ruby 2.3, you can write this

```ruby
if result.user.google_user_info&.google_user_id != google_hash[:google_user_id]
```

;)

Reply:
Oh, that's nicer... though unexpectedly (for me initially) I found:

`result.user.google_user_info&.google_user_id != google_hash[:google_user_id]` is `true` if `result.user.google_user_info` is nil, so I'd need to change all 3 checks